src/jit/codegenlegacy.cpp

```diff
@@ -19216,7 +19216,7 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                     noway_assert(helperNum != CORINFO_HELP_UNDEF);
 
 #ifdef FEATURE_READYTORUN_COMPILER
-                    if (call->gtEntryPoint.addr != NULL)
+                    if (call->gtEntryPoint.addr != NULL && compiler->IsTargetAbi(CORINFO_CORERT_ABI))
```

Comment:
The non-legacy backend has this condition without special casing it for CoreRT ABI. What is special about the legacy backend that it needs this condition?